### PR TITLE
Add Ghoul

### PR DESCRIPTION
**Pull request by weluvgoatz**
This is a new enemy for the ghost forest.

### CODE REVIEW

File: src/badguy/ghoul.cpp [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  PathObject(),
+  m_mystate(STATE_IDLE),
+  m_flyspeed(),
+  m_track_range()
+{
+  reader.get("flyspeed", m_flyspeed, FLYSPEED);
+  reader.get("track-range", m_track_range, TRACK_RANGE);
+  
+  bool running;
+  if ( !reader.get("running", running)) running = false;
```
Comment (tobbi) [inline src/badguy/ghoul.cpp]:
```c++ 
reader.get("running", running, false);
```
should suffice.

File: src/badguy/ghoul.cpp [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+Ghoul::activate()
+{
+  if (Editor::is_active())
+    return;
+}
+
+void
+Ghoul::deactivate()
+{
+  switch (m_mystate) {
```
Comment (tobbi) [inline src/badguy/ghoul.cpp]: I don't think you have to list all the states here.

You can simply just list the state that does something and at the end do
```c++
default:
  break;
```

File: src/badguy/ghoul.cpp [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+Ghoul::active_update(float dt_sec)
+{
+  if (Editor::is_active() && get_path() && get_path()->is_valid()) {
+    get_walker()->update(dt_sec);
+    set_pos(get_walker()->get_pos());
+    return;
+  }
+
+  auto player = get_nearest_player();
+  if (!player) return;
```
Comment (tobbi) [inline src/badguy/ghoul.cpp]: Please put the ```return;``` in a new line.

File: src/badguy/ghoul.hpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+#include "badguy/badguy.hpp"
+#include "object/path_object.hpp"
+
+class Ghoul final : public BadGuy,
+                    public PathObject
+{
+public:
+  Ghoul(const ReaderMapping& reader);
+  virtual std::string get_class() const override { return "ghoul"; }
```
Comment (tobbi) [inline src/badguy/ghoul.hpp]: I do not think we want to subclass Ghoul at any point. Please remove the "virtual" specifier from all the methods.

**Comment by weluvgoatz** [comment]
Thank you @tobbi.  It is fixed now.

**Comment by HybridDog** [comment]
I've added a small style change commit.

File: src/badguy/ghoul.hpp [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  float m_track_range;
+  
+private:
+  Ghoul(const Ghoul&) = delete;
+  Ghoul& operator=(const Ghoul&) = delete;
+};
+
+#endif
+
+/* EOF */
```
Comment (HybridDog) [inline src/badguy/ghoul.hpp]: Here's a missing newline.
https://github.com/SuperTux/supertux/blob/master/CODINGSTYLE.md#file-formating

Comment (weluvgoatz) [reply src/badguy/ghoul.hpp]: fixed

File: src/badguy/ghoul.cpp (hunk 2 above)
Comment (weluvgoatz) [reply src/badguy/ghoul.cpp]: fixed

Comment (HybridDog) [reply src/badguy/ghoul.cpp]: I think that you could also use a simple if statement to make it even more simple.
```C++
if (m_mystate == STATE_TRACKING)
  mystate = STATE_IDLE;
```

**Comment by HybridDog** [comment]
https://travis-ci.org/SuperTux/supertux/jobs/590824166#L3969

File: src/badguy/ghoul.hpp (hunk 4 above)
Comment (HybridDog) [reply src/badguy/ghoul.hpp]: I think you need to keep the `override`s
https://travis-ci.org/SuperTux/supertux/jobs/590824166#L3969

File: src/badguy/ghoul.hpp [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  void finish_construction();
+
+  void activate();
+  void deactivate();
+  void active_update(float dt_sec);
+  
+  void goto_node(int node_no);
+  void set_state(const std::string& state);
+  void start_moving();
+  virtual void stop_moving();
```
Comment (tobbi) [inline src/badguy/ghoul.hpp]: Remove the "virtual" from this, please.

Comment (weluvgoatz) [reply src/badguy/ghoul.hpp]: (hopefully) fixed

**Pull request merged**